src/libraries/System.Text.Json/Common/JsonSeparatorNamingPolicy.cs

```diff
@@ -21,149 +28,148 @@ public sealed override string ConvertName(string name)
                 ThrowHelper.ThrowArgumentNullException(nameof(name));
             }
 
-            // Rented buffer 20% longer that the input.
-            int rentedBufferLength = (12 * name.Length) / 10;
-            char[]? rentedBuffer = rentedBufferLength > JsonConstants.StackallocCharThreshold
-                ? ArrayPool<char>.Shared.Rent(rentedBufferLength)
-                : null;
-
-            int resultUsedLength = 0;
-            Span<char> result = rentedBuffer is null
-                ? stackalloc char[JsonConstants.StackallocCharThreshold]
-                : rentedBuffer;
+            return ConvertNameCore(_separator, _lowercase, name);
+        }
 
-            void ExpandBuffer(ref Span<char> result)
-            {
-                char[] newBuffer = ArrayPool<char>.Shared.Rent(result.Length * 2);
+        private static string ConvertNameCore(char separator, bool lowercase, string name)
+        {
+            Debug.Assert(name != null);
 
-                result.CopyTo(newBuffer);
+            char[]? rentedBuffer = null;
 
-                if (rentedBuffer is not null)
-                {
-                    result.Slice(0, resultUsedLength).Clear();
-                    ArrayPool<char>.Shared.Return(rentedBuffer);
-                }
+            // Rented buffer 20% longer that the input.
+            int initialBufferLength = (12 * name.Length) / 10;
```

Comment:
Why not the simpler:
```suggestion
            int initialBufferLength = (int)(name.Length * 1.2);
```
?

Reply:
I would guess it was trying to avoid floats and use integral division only? I don't think it matters much.